src/components/lyne-button/styles/lyne-button.component.scss

```diff
@@ -82,22 +51,52 @@
   --color-hover-text: var(--color-iron-default);
 }
 
-.button--small {
-  --border-width: var(--border-width-1x);
-  --min-height: var(--size-form-element-small-min-height);
-  --padding-block: var(--spacing-fixed-2x);
-  --padding-inline: var(--spacing-fixed-5x);
+.button--transparent {
+  --color-active-background: var(--color-cloud-default);
+  --color-active-border: var(--color-cloud-default);
+  --color-active-text: var(--color-black-default);
+  --color-default-background: transparent;
+  --color-default-border: transparent;
+  --color-default-text: var(--color-black-default);
+  --color-disabled-background: transparent;
+  --color-disabled-border: transparent;
+  --color-disabled-text: var(--color-granite-default);
+  --color-hover-background: var(--color-milk-default);
+  --color-hover-border: var(--color-milk-default);
+  --color-hover-text: var(--color-black-default);
+}
+
+.button--transparent-negative {
+  --color-active-background: var(--color-iron-default);
+  --color-active-border: var(--color-iron-default);
+  --color-active-text: var(--color-white-default);
+  --color-default-background: transparent;
+  --color-default-border: transparent;
+  --color-default-text: var(--color-white-default);
+  --color-disabled-background: transparent;
+  --color-disabled-border: transparent;
+  --color-disabled-text: var(--color-granite-default);
+  --color-hover-background: var(--color-charcoal-default);
+  --color-hover-border: var(--color-charcoal-default);
+  --color-hover-text: var(--color-white-default);
+}
 
-  // TODO: missing design tokens
-  --font-size: 12;
+.button--size-m {
+  @include buttonSizeMVariables;
 }
 
 .button--icon-only {
   --padding-inline: 0;
   --icon-margin-inline-end: 0;
 
-  width: pxToRem(var(--min-height));
+  width: pxToRem(var(--min-height-hover, var(--min-height)));
   justify-content: center;
+
+  &.button:disabled::after {
```

Comment:
to set the strike-through width, wouldn't be simple doing something like this?

`
width: calc(100% - buttonHorizontalPaddingVariable);
`

instead of creating a dedicated variable (--line-through-icon-button-padding-inline) and using inset?

Reply:
I completly rewrote it because I found out that in icon variant the width is always 24px. Thanks for the comment, I could simplify it ;)